spec/controllers/sessions_controller_spec.rb

```diff
@@ -0,0 +1,101 @@
+require 'spec_helper'
+
+describe SessionsController do
+  context "User not logged in:" do
+    describe "GET new:" do
+      it "Renders the new template" do
+        get :new
+        expect(response).to render_template :new
+      end
+    end
+
+    describe "POST create:" do
+      context "User exists and authenticates:" do
+        before do
+          user = Fabricate(:user)
+          post :create, email: user.email, password: user.password
+        end
+
+        it "Should set sessions to user_id" do
+          expect(session[:user_id]).to eq(User.first.id)
+        end
+
+        it "Should redirect to home" do
+          expect(response).to redirect_to root_path
+        end
+      end
+
+      context "User login invalid:" do
+        before do
+          @user = Fabricate(:user)
```

Comment:
Thanks for letting me know! It's certainly a subtle advantage I wouldn't have realized myself.

http://stackoverflow.com/questions/14116613/when-is-using-instance-variables-more-advantageous-than-using-let